src/main/java/com/app/tacoLoco/TacoService.java

```
@@ -1,49 +1,83 @@
 package com.app.tacoLoco;
 
+import java.text.DecimalFormat;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 import org.springframework.stereotype.Service;
 
 @Service
 public class TacoService {
-	
-	private List<Taco> newOrder = new ArrayList<>(
-	); 
-
-	public Object getTacoPrices() {
-		
-		for (int i = 0; i < Taco.orderItems.size(); i++) { 
-			
-			if (Taco.orderItems.get(i).toString() == "veggieTaco") {
-			   Taco.orderItemPrices.add((float) 2.5);
+
+	static List<Taco> newOrder = new ArrayList<>();
+
+	static List<Double> tacoPrices = new ArrayList<>();
+
+	public static Object getTacoPrices(ArrayList<String> orderItems) {
+
+		for (int i = 0; i < orderItems.size(); i++) {
+
+			Double veggieTaco = (double) 2.5;
+			Double beefTaco = (double) 3;
+			Double chickenTaco = (double) 3;
+			Double chorizoTaco = (double) 3.5;
+
+			String item = orderItems.get(i).toLowerCase().replaceAll("\\s+","");
+
+			if (item.equals("veggietaco") == true) {
+			    tacoPrices.add(veggieTaco);
+		    }
+
+			else if (item.equals("beeftaco") == true) {
+				tacoPrices.add(beefTaco);
+			}
+
+			else if (item.equals("chickentaco") == true) {
+				tacoPrices.add(chickenTaco);
+			}
+
+			else if (item.equals("chorizotaco") == true) {
+				tacoPrices.add(chorizoTaco);
+			}
+
+			else {
+				return "One of your items was invalid. Please revise your order and submit it again";
```

We may need to talk more about this. But if the client were to submit something that they are not supposed to we would need to indicate to them that they have submitted what we call a "bad request". One of the things we do to let the client know its a bad request is to send them back a response code of 404 on their HTTP response. If we send them back this message (which is still very meaningful), it wouldn't suffice because the response code would still be 200 and that would be misleading. So for starters I would suggest reading into HTTP status codes.
As for this specific use case, because Spring is the best they would do most of that handling for us. We would just need to throw an exception here and spring would translate that into a corresponding response code on the HTTP response.